src/polygon/src/polygon.cpp

```diff
@@ -155,6 +158,46 @@ void CheckGeometry(const std::vector<glm::ivec3> &triangles,
                                   vertPos[tri[2]], precision) >= 0;
                      }),
          geometryErr, "triangulation is not entirely CCW!");
+
+  std::vector<std::pair<glm::vec2, glm::vec2>> edges(3 * triangles.size());
+  std::for_each(
+#if MANIFOLD_PAR == 'T' && __has_include(<pstl/glue_execution_defs.h>)
+      std::execution::par,
+#endif
+      countAt(0ul), countAt(triangles.size()), [&](size_t i) {
+        edges[3 * i] =
+            std::make_pair(vertPos[triangles[i].x], vertPos[triangles[i].y]);
+        edges[3 * i + 1] =
+            std::make_pair(vertPos[triangles[i].y], vertPos[triangles[i].z]);
+        edges[3 * i + 2] =
+            std::make_pair(vertPos[triangles[i].z], vertPos[triangles[i].x]);
+      });
+  std::mutex mutex;
+  std::vector<std::pair<size_t, size_t>> overlappingPairs;
+  auto p = [&](size_t i) {
+    return std::all_of(countAt(i + 1), countAt(edges.size()), [&](size_t j) {
+      if (intersect(edges[i].first, edges[i].second, edges[j].first,
+                    edges[j].second, precision)) {
+        mutex.lock();
+        overlappingPairs.push_back(std::make_pair(i, j));
+        mutex.unlock();
+        return true;
```

Comment:
I found that I actually flipped the true/false. The actual check takes much longer even with some optimization, maybe I should hide it after another parameter. Will fix.

Reply:
When this is fixed, Sponge4 takes ~15s to run, which is pretty slow...

Reply:
How much difference does removing the intersection checks make now? Those look like they would be pretty slow...